port/osx/omrintrospect.c

```diff
@@ -274,22 +303,14 @@ setupNativeThread(J9ThreadWalkState *state, thread_context *sigContext)
 	return rc;
 }
 
-static int
+static int32_t
 getThreadContext(J9ThreadWalkState *state)
 {
-	int ret = 0;
-	char buffer[1];
+	int32_t ret = 0;
 	PlatformWalkData *data = (PlatformWalkData *)state->platform_data;
 	thread_port_t thread = data->threadList[data->threadIndex];
 
-	/* Create a pipe to allow the resumed thread to report it has completed
-	 * sending its context info.
-	 */
-	ret = pipe(pipeFileDescriptor);
-
-	if (0 == ret) {
-		ret = pthread_kill(pthread_from_mach_thread_np(thread), SUSPEND_SIG);
-	}
+	ret = pthread_kill(pthread_from_mach_thread_np(thread), SUSPEND_SIG);
```

Comment:
`pthread_from_mach_thread_np` returns null on error and if `pthread_kill` receives a null thread it simply returns an error code
```
if (!_pthread_is_valid(th, &kport)) {
	return ESRCH;
}
```
Is the additional null check necessary?

Reply:
It may not be absolutely necessary, but IMO it's much cleaner (and removes the dependency on what `pthread_kill(NULL)` means). On Linux it may 'cause a segmentation fault' (see [pthread_kill(3)](https://man7.org/linux/man-pages/man3/pthread_kill.3.html)); it's safer to assume such bad things could happen on macOS as well.